### PR TITLE
Improved gpio handling for buttons and status LEDs

### PR DESCRIPTION
**Pull request by MuratUrsavas**
**Issue**

#172 

**How**
If a user button or status led definition is missing in a variant definition, that part of the code is passed gracefully now.

**Checklist**

- [ ] Tests added
- [x] Cleaned up commit history (rebase!)
- [ ] Documentation added
- [x] Thought about variable and method names

### CODE REVIEW

**Comment by shawaj** [comment]
@MuratUrsavas in hm-pyhelper would we put:
`'STATUS': False,` and `'BUTTON': False,` if they didn't exist? Or just leave them out entirely? 

Or is either ok?

**Comment by MuratUrsavas** [comment]
@shawaj We have to left them out as otherwise it would return something (False), and it would break current code and give exceptions. It depends on missing keys.

**Comment by shawaj** [comment]
I'll put a note on that repo to update the readme to reflect that

**Review by shawaj — APPROVED** [review]

File: gatewayconfig/gatewayconfig_app.py [hunk 1]
```diff
@@ -125,28 +131,39 @@ def init_gpio(self, variant):
...
+                self.user_button = Button(btn, hold_time=USER_BUTTON_HOLD_SECONDS)
+            led = self.get_status_led_gpio()
+            if led is not None:
+                self.status_led = LED(led)
         elif is_rockpi():
-            self.user_button = MraaButton(self.get_button_pin(), hold_seconds=USER_BUTTON_HOLD_SECONDS)
-            self.user_button.start()
-            self.status_led = MraaLED(self.get_status_led_pin())
+            btn = self.get_button_pin()
+            if btn is not None:
```
Comment (pritamghanghas) [inline gatewayconfig/gatewayconfig_app.py:142]: if btn: 
would have been the same.

**Pull request merged**